packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition_tradeoptions.js

```diff
@@ -375,22 +375,28 @@ Blockly.Blocks.trade_definition_tradeoptions = {
                     return false;
                 }
 
-                const { min, max } = this.durations.find(d => d.unit === this.selected_duration);
-                const is_valid_duration = input_number >= min && input_number <= max;
-
-                if (min === max) {
-                    this.error_message = localize(
-                        'Duration value is not allowed. To run the bot, please enter {{min}}.',
-                        { min }
-                    );
-                } else {
-                    this.error_message = localize(
-                        'Duration value is not allowed. To run the bot, please enter a value between {{min}} to {{max}}.',
-                        { min, max }
-                    );
+                const duration = this.durations.find(d => d.unit === this.selected_duration);
+
+                if (duration) {
+                    const { min, max } = duration;
+                    const is_valid_duration = input_number >= min && input_number <= max;
+
+                    if (min === max) {
+                        this.error_message = localize(
+                            'Duration value is not allowed. To run the bot, please enter {{min}}.',
+                            { min }
+                        );
+                    } else {
+                        this.error_message = localize(
+                            'Duration value is not allowed. To run the bot, please enter a value between {{min}} to {{max}}.',
+                            { min, max }
+                        );
+                    }
+
+                    return !is_valid_duration;
                 }
 
-                return !is_valid_duration;
+                return false;
```

Comment:
What does this `return false` do?

Reply:
It's a bit ambiguous, but it tells the validator to not consider this block as required. This code is within the `getRequiredValueInputs`, returning `true` will tell the validator (see `valueInputLimitationsListener` in `dbot.js`) whether to highlight the block or not.